### DISCUSSION TITLE
Sigma x and Sigma y for gaus blur

### DISCUSSION THREAD

**Discussion by Ruchika-K** [libvips/libvips#2961] [Q&A] [answered]
Is it possible to apply gaus blurring using a non square kernel i,e, use both Sigma x and Sigma y.
Basically I want to apply something similar to opencv - 
SigmaX ,sigmaY- https://docs.opencv.org/4.x/d4/d86/group__imgproc__filter.html#gac05a120c1ae92a6060dd0db190a61afa

**Comment by jcupitt** [comment, marked as answer]
Hi @Ruchika-K,

Sure, use `gaussmat` to make two 1D masks (set `separable`) with different sigma, rotate one them 90 degrees, and convolve twice.

In eg. python (untested):

```python
def gblur2(input, sigma1, sigma2):
    m1 = pyvips.Image.gaussmat(sigma1, 0.1, separable=True)
    m2 = pyvips.Image.gaussmat(sigma2, 0.1, separable=True).rot90()
    return input.conv(m1).conv(m2)
```

**Reply by Ruchika-K** [reply]
Thanks!  Will try this out 
Is there a specific reason to take min ampl as 0.1 ?
I have been using opencv till now and there is no such attribute there. What value should I set for ampl if I just have the kernel size or sigmas(in terms of opencv gaus)

**Reply by jcupitt** [reply]
0.2 is fine for almost all uses, 0.1 is super-accurate.

You can see the mask like this:

```
$ vips gaussmat x.csv 2 0.1 
$ more x.csv 
0	1	2	2	3	2	2	1	0
1	2	4	6	6	6	4	2	1
2	4	7	11	12	11	7	4	2
2	6	11	16	18	16	11	6	2
3	6	12	18	20	18	12	6	3
2	6	11	16	18	16	11	6	2
2	4	7	11	12	11	7	4	2
1	2	4	6	6	6	4	2	1
0	1	2	2	3	2	2	1	0
```
